Bounty description templates - Finished w/ updated data

rej156 opened the pull request.

corwinharrell commented:
### QA Items:

[] The `MarkdownKey` component used to be attached to instances of the markdown editor for creating a bounty and fulfilling a submission. At the moment, they are absent:

<img width="593" alt="Screen Shot 2019-08-28 at 8 19 54 PM" src="https://user-images.githubusercontent.com/5934188/63901104-39685f00-c9d1-11e9-9fed-3b8226d8d137.png">

[] For the `Default` template, the helper text for the template is placeholder text. How about something like:

> The default description template provides a simple and multi-purpose template for writing well structured and clear instructions for potential fulfillers.

[] If I click into and out of the `Category` select dropdown while it has categories pre-selected, those categories disappear.

[] For some previews for description templates that are especially long, the preview modal gets positioned behind the fixed header. In these modal instances, we should add some top margin or padding to the modal to prevent it from beginning behind the header.

<img width="867" alt="Screen Shot 2019-08-28 at 11 07 10 PM" src="https://user-images.githubusercontent.com/5934188/63907209-9a4f6180-c9e8-11e9-8d27-013d4b3a0485.png">

[] I believe that there were additional commits or change made to the discard changes modal (when leaving long forms without saving). There was an icon added and some changes to styling. Were these possibly left behind in a different branch?

[] I think we should introduce an appropriate max-height for the `Textbox` component. In order for the `MarkdownKey` to be useful, it needs to remain in the viewport in the majority of cases.

### Additional thoughts for consideration:

- @rej156 I noticed that in implementing the templates in the json file, you included breaks after each guiding question in the template. I'm interested to see this (not necessarily opposed). Do you think that makes it easier to scan/absorb?

- In the original mocks, we had planned to add an icon within the dropdown menu for each template label. I know that we are using a third party component for the dropdown select at the moment, and I couldn't figure out an easy way to include icons as a part of the labels inside of the dropdown. @rej156 upon a quick inspection, do you see a low-lift way of adding this or do you think it's not worth the effort at the moment? I would think that this is something we wouldn't wan to spend _too_ much time on.

rej156 commented on src/i18n/en_US.json:
```diff
@@ -320,20 +320,89 @@
+        "default": {
+          "label": "Default",
+          "about":
+            "This is a short description of the template that has been selected above. It provides some insight into how this template might be used, in addition to some potential example use cases.",
```
https://hjson.org/try.html 

Just found this

rej156 commented:
> * [x]  The MarkdownKey component used to be attached to instances of the markdown editor for creating a bounty and fulfilling a submission. At the moment, they are absent:

Re-added it back in, had to backtrack multiple commits prior to mine :S 

> * [x]  For the Default template, the helper text for the template is placeholder text. How about something like:

Done 💯 

> * [x]  If I click into and out of the `Category` select dropdown while it has categories pre-selected, those categories disappear.

Made it concat, add to existing

> * [ ]  For some previews for description templates that are especially long, the preview modal gets positioned behind the fixed header. In these modal instances, we should add some top margin or padding to the modal to prevent it from beginning behind the header.

Yo could you re-qa this, added a modal-open class to the page-header and slowed down the css transition, hard to reproduce consistently.

> * [ ]  I believe that there were additional commits or change made to the discard changes modal (when leaving long forms without saving). There was an icon added and some changes to styling. Were these possibly left behind in a different branch?

Err not a clue :((

> * [x]  I think we should introduce an appropriate max-height for the `Textbox` component. In order for the `MarkdownKey` to be useful, it needs to remain in the viewport in the majority of cases.
> 

Added 50em maxHeight prop


> ### Additional thoughts for consideration:
> * @rej156 I noticed that in implementing the templates in the json file, you included breaks after each guiding question in the template. I'm interested to see this (not necessarily opposed). Do you think that makes it easier to scan/absorb?

I'm prone to text keyboard shortcuts, added it because it feels easier to read + I inline edit mentally after selecting the whole line via cmd+shift+left and just typing in the answer

> * In the original mocks, we had planned to add an icon within the dropdown menu for each template label. I know that we are using a third party component for the dropdown select at the moment, and I couldn't figure out an easy way to include icons as a part of the labels inside of the dropdown. @rej156 upon a quick inspection, do you see a low-lift way of adding this or do you think it's not worth the effort at the moment? I would think that this is something we wouldn't wan to spend _too_ much time on.

Added a commit, found a way to do this 👍 Just replicate

corwinharrell commented:
> Yo could you re-qa this, added a modal-open class to the page-header and slowed down the css transition, hard to reproduce consistently.

The fix looks good!

> Err not a clue :((

Re: The warning dialogue. No worries at all, I'll add that stuff back in.

> I'm prone to text keyboard shortcuts, added it because it feels easier to read + I inline edit mentally after selecting the whole line via cmd+shift+left and just typing in the answer

Fair enough. I just wanted to hear why you preferred this formatting. I think those are good reasons so I'll leave the formatting as-is 👍 

> Added a commit, found a way to do this 👍 Just replicate

Awesome! I'll go through and add the appropriate icons.

corwinharrell commented:
@rej156 This looks just about ready. As I mentioned in slack, I'm just going to push a few commits for a few small loose ends before merging 👍

corwinharrell commented on src/fontAwesome.js:
```diff
@@ -78,8 +79,25 @@ import userFriends from '@fortawesome/fontawesome-pro-regular/faUserFriends';
+//
+// import fileSignature from '@fortawesome/fontawesome-pro-regular/faFileSignature';
+// import pencilPaintbrush from '@fortawesome/fontawesome-pro-regular/faPencilPaintbrush';
+// import voteYea from '@fortawesome/fontawesome-pro-regular/faVoteYea';
```
@rej156 I'm having some trouble figuring out why these icons in particular are throwing errors when I try to import them. Uncomment them to see what I mean. Do you think you could take a look? Curious to hear what you think the issue might be.

rej156 replied:
Just cooking haddock for dinner ATM, will check after bae

corwinharrell replied:
No you won't! Enjoy your Sunday evening! I'm just wrapping up a few things because I'm heading out on vacation this week :) Can wait till Monday

corwinharrell commented:
@rej156 @mbeylin once the icon issue I mentioned to Eric above is fixed, and the description content for the Translations template has been added (Mark I tagged you in a Notion comment for this), this should be ready to merge and deploy.

@rej156 I made a few small formatting changes to a couple of the templates, but for the most part, your formatting remains intact.

rej156 commented:
Updated the font awesome icons usage around the dApp
@mbeylin it just needs that translation template now

corwinharrell commented:
@rej156 I've updated description for the translation bounty using the content that Simona provided. That should mean that we can 🚢 this! I'll let you be the one to rebase master and merge.

rej156 commented:
:shipit:

mbeylin commented on src/containers/CreateBounty/index.js:
```diff
@@ -282,8 +282,7 @@ const mapStateToProps = (state, router) => {
       categories: categories,
-      description:
-        draftBounty.description || intl.get('components.editor.default'),
+      description: intl.get('components.editor.default'),
```
@rej156 incase you're curious, this was the line that messed up the bounty description editing
